### PR TITLE
MGDOBR-810: Add Cloud Provider and Cloud Region options to the create bridge dialog

### PR DESCRIPTION
**Pull request by kelvah**
JIRA: https://issues.redhat.com/browse/MGDOBR-810

This PR depends on:
- https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/899

With this PR we are adding the `Cloud Provider` and `Cloud Region` options inside the bridge creation dialog.
We only support one provider and one region, so the user can't change the default selection for both.

Steps to test this:
- run the application 
- click on the "Create Smart Events Instance" button
- the new provider and region fields are present and their default values are set
- provide a name for the instance
- click on the "Create Smart Events Instance" button inside the dialog
- the bridge is created and the payload of the request includes cloud provider and cloud region

If no regions are available (meaning the region retrieved has the `enabled` property set to `false`) the entire form is disabled and  a warning message is shown. You can test this scenario by using the mocked APIs, [manipulating mocks](https://github.com/kelvah/sandbox-ui/blob/a93d5ff3489c2cdf167cd60585b419165608bc85/mocked-api/cloudProvidersData.ts#L25) and setting `enabled=false` to the only existing region.

### CODE REVIEW

**Review by manstis — APPROVED** [review]
LGTM. A couple of minor unrelated comments.

**Review by vpellegrino — APPROVED** [review]
Overall the PR looks good to me

**Comment by jomarko** [comment]
Waiting with my review until https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/899 is merged

**Review by sjamboro — APPROVED** [review]
LGTM ... there are several blocks of code which I cannot test via UI but some of them are covered by components tests and there are new JIRA issues which will solve it. For example:
https://issues.redhat.com/browse/MGDOBR-927

The main feature is that the Creation form contains the new fields with the default values. The values are loading from REST API when an user clicks on the `Create SE Instance` button.

File: src/hooks/useBridgesApi/useCreateBridgeApi.ts [hunk 1]
```diff
@@ -8,7 +8,14 @@ import { useState } from "react";
 import { useSmartEvents } from "@contexts/SmartEventsContext";
 
 export function useCreateBridgeApi(): {
-  createBridge: (bridgeRequest: BridgeRequest) => void;
+  // revert to BridgeRequest when the corresponding BE PR will be merged
```
[the remark this answers is not in the document]
Comment (kelvah) [reply src/hooks/useBridgesApi/useCreateBridgeApi.ts]: Fixed now.

File: openapi/generated/errorHelpers.ts [hunk 2]
```diff
@@ -1,5 +1,12 @@
 import { AxiosError } from "axios";
-import { ErrorListResponse, ErrorResponse } from "@openapi/generated/model";
+import { ErrorListResponse,  ModelError } from "@openapi/generated/model";
+
+// @TODO Temporary workaround because the Error interface is not being
+// replaced with ModelError everywhere. See https://issues.redhat.com/browse/MGDOBR-979
+// for more details
+interface ErrorListResponseFixed {
+    items?: Array<ModelError>
+}
```
Comment (kelvah) [inline openapi/generated/errorHelpers.ts:9]: Workaround until we find a proper fix

**Review by kelvah — COMMENTED** [review]
I took the chance to delete the two errors hooks (`useGetErrorApi.ts` and `useGetErrorsApi.ts`) instead of fixing them. 
We never used them and I see no future use for them either.

**Pull request merged**